src/http/api/resources/repo.js

```diff
@@ -1,9 +1,25 @@
 'use strict'
 
-exports.gc = async (request, h) => {
-  const { ipfs } = request.server.app
-  await ipfs.repo.gc()
-  return h.response()
+const Joi = require('joi')
+
+exports.gc = {
+  validate: {
+    query: Joi.object().keys({
+      'stream-errors': Joi.boolean().default(false)
+    }).unknown()
+  },
+
+  async handler (request, h) {
+    const streamErrors = request.query['stream-errors']
+    const { ipfs } = request.server.app
+    const res = await ipfs.repo.gc()
+
+    const filtered = res.filter(r => !r.err || streamErrors)
+    const response = filtered.map(r => {
+      return { Err: r.err, Key: !r.err && { '/': r.cid } }
```

Comment:
`r.err` is an `Error`? Does it need to be changed to `r.err.message`?

`r.cid` this should be a CID instance right? Does it need a `toString()`?